File: seed/csharp-model/exhaustive/Types/Object/NestedObjectWithRequiredField.cs
using System.Text.Json.Serialization
using test.Types

namespace test.Types

public class NestedObjectWithRequiredField
{
    [JsonPropertyName("string")]
    public string String { get; init; }
    [JsonPropertyName("NestedObject")]
    public ObjectWithOptionalField NestedObject { get; init; }
}

File: seed/php-model/exhaustive/src/Types/Object/NestedObjectWithRequiredField.php
<?php

namespace Seed\Types\Object;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class NestedObjectWithRequiredField extends JsonSerializableType
{
    /**
     * @var string $string
     */
    #[JsonProperty('string')]
    public string $string;

    /**
     * @var ObjectWithOptionalField $nestedObject
     */
    #[JsonProperty('NestedObject')]
    public ObjectWithOptionalField $nestedObject;

    /**
     * @param array{
     *   string: string,
     *   nestedObject: ObjectWithOptionalField,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->string = $values['string'];$this->nestedObject = $values['nestedObject'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
